### PR TITLE
Documentation for stype/ltype

### LINKED ISSUE #2628: Support extraction of docstrings from python
Currently `.. xfunction::` directive only knows how to extract docs from C++ source files.
However, some documentation lives in Python, and we need to be able to incorporate it into the docs.

### LINKED ISSUE #2633: Documentation for .build_info didn't render on readthedocs
See https://datatable.readthedocs.io/en/latest/api/dt/build_info.html (currently it is blank)

### PR DESCRIPTION
**Pull request by st-pasha**
- Added documentation for classes `stype` and `ltype`;
- Implemented extracting docstrings from python code;
- Allow possibility to have a method/function without the source (this allows the docs for `.build_info` to build properly).

WIP for #2586
Closes #2628
Closes #2633

### CODE REVIEW

File: src/datatable/sphinxext/xfunction.py [hunk 1]
```diff
@@ -1112,6 +1096,183 @@ def _generate_body(self):
...
+def locate_python_variable(name, lines):
+    """
+    Find declaration of a variable `name` within python source `lines`.
+
+    Returns a tuple (istart, iend) such that `lines[istart:iend]` would
+    contain the lines where the variable is declared. Variable declaration
+    must have the form of an assignment, i.e. "{name} = ...". Other ways
+    of declaring a variable are not supported.
+
+    A ValueError is raised if variable declaration cannot be found.
```
Comment (oleksiyskononenko) [inline src/datatable/sphinxext/xfunction.py:1112]: Back-ticks around `ValueError`.

File: src/datatable/sphinxext/xfunction.py [hunk 2]
```diff
@@ -1112,6 +1096,183 @@ def _generate_body(self):
...
+    """
+    Find declaration of a class/function `name` in python source `lines`.
+
+    Returns a tuple (istart, iend) such that `lines[istart:iend]` would
+    contain the lines where the class/function is declared. The body of
+    the class/function starts with "(class|def) {name}..." and ends
+    at a line with the indentation level equal or smaller than the
+    level of the declaration start.
+
+    A ValueError is raised if variable declaration cannot be found.
```
Comment (oleksiyskononenko) [inline src/datatable/sphinxext/xfunction.py:1136]: Back-ticks around "ValueError".

**Review by oleksiyskononenko — APPROVED** [review]

File: docs/api/ltype.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    -------
+    .. list-table::
+        :widths: auto
+        :class: api-table
+
+        * - :meth:`ltype(x) <datatable.ltype.__new__>`
+          - Find ltype corresponding to value ``x``.
+
+        * - :attr:`.stypes <datatable.ltype.stypes>`
+          - The list of :class:`stype`s that correspond to this ltype.
```
[the remark this answers is not in the document]
Comment (st-pasha) [reply docs/api/ltype.rst:27]: the verb belongs to "stypes", which is plural. So it's "The list of (stypes that correspond to this ltype)".

File: docs/api/ltype.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    Examples
+    --------
+    >>> dt.ltype.bool
+    ltype.bool
+    >>> dt.ltype("int32")
+    ltype.int
+
+    For each ltype, you can find the set of stypes that correspond to it:
```
[the remark this answers is not in the document]
Comment (st-pasha) [reply docs/api/ltype.rst:36]: (same)

File: src/datatable/sphinxext/xfunction.py (hunk 2 above)
Comment (st-pasha) [reply src/datatable/sphinxext/xfunction.py:1136]: This is an internal (not user-visible) documentation so I only use backticks to disambiguate code from regular text. For example, "source lines" is just a description, whereas "source \`lines\`" specifically refers to the variable `lines`

Comment (oleksiyskononenko) [reply src/datatable/sphinxext/xfunction.py:1136]: Yes, but at some point the internal docs may go public fully or partially, like in the case of the new options, etc.

**Pull request merged**